php-checks/src/main/java/org/sonar/php/checks/FormattingStandardCheck.java

```diff
@@ -19,69 +19,45 @@
  */
 package org.sonar.php.checks;
 
-import com.sonar.sslr.api.AstNode;
 import org.sonar.api.server.rule.RulesDefinition;
 import org.sonar.check.Priority;
 import org.sonar.check.Rule;
 import org.sonar.check.RuleProperty;
-import org.sonar.php.api.PHPPunctuator;
 import org.sonar.php.checks.formatting.ControlStructureSpacingCheck;
 import org.sonar.php.checks.formatting.CurlyBraceCheck;
 import org.sonar.php.checks.formatting.ExtendsImplementsLineCheck;
+import org.sonar.php.checks.formatting.FormattingCheck;
 import org.sonar.php.checks.formatting.FunctionSpacingCheck;
 import org.sonar.php.checks.formatting.IndentationCheck;
 import org.sonar.php.checks.formatting.NamespaceAndUseStatementCheck;
 import org.sonar.php.checks.formatting.PunctuatorSpacingCheck;
-import org.sonar.php.parser.PHPGrammar;
+import org.sonar.plugins.php.api.tree.ScriptTree;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
 import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
 import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
-import org.sonar.squidbridge.checks.SquidCheck;
-import org.sonar.sslr.grammar.GrammarRuleKey;
-import org.sonar.sslr.parser.LexerlessGrammar;
-
-import javax.annotation.Nullable;
-
-import java.util.Arrays;
 
 @Rule(
-  key = "S1808",
+  key = FormattingStandardCheck.KEY,
   name = "Source code should comply with formatting standards",
   priority = Priority.MINOR,
   tags = {Tags.CONVENTION, Tags.PSR2})
 @SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
 @SqaleConstantRemediation("1min")
-public class FormattingStandardCheck extends SquidCheck<LexerlessGrammar> {
-
-  private static final GrammarRuleKey[] CLASS_AND_FUNCTION = {
-    PHPGrammar.CLASS_DECLARATION,
-    PHPGrammar.INTERFACE_DECLARATION,
-    PHPGrammar.TRAIT_ADAPTATIONS,
-    PHPGrammar.METHOD_DECLARATION,
-    PHPGrammar.FUNCTION_DECLARATION
-  };
-
-  private static final GrammarRuleKey[] CONTROL_STRUCTURE = {
-    PHPGrammar.IF_STATEMENT,
-    PHPGrammar.ELSEIF_CLAUSE,
-    PHPGrammar.ELSE_CLAUSE,
-    PHPGrammar.DO_WHILE_STATEMENT,
-    PHPGrammar.WHILE_STATEMENT,
-    PHPGrammar.FOR_STATEMENT,
-    PHPGrammar.FOREACH_STATEMENT,
-    PHPGrammar.SWITCH_STATEMENT,
-    PHPGrammar.TRY_STATEMENT,
-    PHPGrammar.CATCH_STATEMENT,
-    PHPGrammar.FINALLY_STATEMENT
+public class FormattingStandardCheck extends PHPVisitorCheck {
+
+  public static final String KEY = "S1808";
+
+  private static FormattingCheck[] SUB_CHECKS = {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "CLASS_AND_FUNCTION" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1068)